### PR TITLE
Fixes #40

### LINKED ISSUE #40: Pragmas and Block Comments
While the pragma feature is cool, these comments will be retained in the generated code:

```
Block comments, which mirror the syntax for block strings, are preserved in the generated code.
````

From my experiments with coffedoc and the "AST" written out by the coffee compiler, OTOH I think that that all documentation will be associated with the node immediately following that documentation.
Would it not be better to parse that documentation instead of leaking these pragmas into the generated code?

### PR DESCRIPTION
**Pull request by silkentrance**
- fixes #40: pragmas and istanbul ignore directives can now be written as inline comments

### CODE REVIEW

**Comment by jwalton** [comment]
Out of the office today, but I'll have a look at this on Monday for sure.  Thanks!

File: src/utils/codeUtils.coffee [hunk 1]
```diff
@@ -20,3 +20,26 @@ exports.minLocation = (locations) ->
...
         if exports.compareLocations(loc, min) < 0 then min = loc
     return min
+
+# Given a coffee-script source string, returns the string with all single line comments representing
+# pragmas replaced into herecomments, so that they will be preserved when building the AST.
+exports.pragmaToHereComment = (source) ->
+    if !source or source.length is 0 then return ''
+
+    result = []
+    re = /^(\s*)#\s*(!pragma.+|istanbul\s+ignore.+)\s*$/
```
Comment (jwalton) [inline src/utils/codeUtils.coffee]: The only weird corner case here would be something like:

    """
    # !pragma ...
    """

Which, I admit, is unlikely...

**Comment by silkentrance** [comment]
I am no longer developing using coffeescript in favor of babel/es. Therefore I deleted my fork. Will close this as it seems as if it is never going to be integrated anyway.

**Comment by jwalton** [comment]
Sorry Carsten; I made a comment and you never replied to it.  :(

**Comment by jwalton** [comment]
Thanks for all the help you offered on this project.  :)

**Comment by silkentrance** [comment]
@jwalton Sorry, I thought that you provided the correct answer yourself, i.e. "Which, I admit, is unlikely...", which I believe it actually is. Nonetheless, moving onward since I do not think that the new features found in babel/ES will be ported to coffeescript, e.g. decorators and so on.

**Comment by jwalton** [comment]
Yeah, babel and es6 look more and more tempting all the time.  Thanks
again.  :)

-Jason

**Comment by silkentrance** [comment]
You are welcome.

**Pull request closed without merging**